src/codegen.cpp

```diff
@@ -1324,88 +1322,78 @@ void *jl_get_llvmf(jl_function_t *f, jl_tupletype_t *tt, bool getwrapper, bool g
         linfo = jl_get_unspecialized(linfo);
     }
 
-#if defined(USE_ORCJIT) || defined(USE_MCJIT)
-    if (linfo->functionObjects.functionObject == NULL && linfo->functionObjects.specFunctionObject == NULL) {
-        jl_compile_linfo(linfo, NULL);
-    }
-    if (getdeclarations) {
-        JL_GC_POP();
-        if (getwrapper || linfo->functionObjects.specFunctionObject == NULL)
-            return linfo->functionObjects.functionObject;
+    if (!getdeclarations) {
+        Function *llvmDecl = nullptr;
```

Comment:
What I mean is that since we are officially swithcing to LLVM 3.7.1 now, we can require a c++11 compiler. Keeping LLVM 3.3 support is useful to compare compiler regressions but not so much about keeping c++98 compatibility.

In any case replacing `nullptr` with `NULL` is a trivial way to fix this usage of `c++11` feature...